core/domain/activity_jobs_one_off.py

```diff
@@ -628,3 +629,110 @@ def reduce(key, values):
+        },
+        'SkillSnapshotMetadataModel': {
+            'parent_model_class': skill_models.SkillModel,
+            'commit_log_model_class': skill_models.SkillCommitLogEntryModel,
+            'id_string_format': 'skill-%s-%s',
+            'id_field': 'skill_id'
+        },
+        'ExplorationRightsSnapshotMetadataModel': {
+            'parent_model_class': exp_models.ExplorationRightsModel,
+            'commit_log_model_class': exp_models.ExplorationCommitLogEntryModel,
+            'id_string_format': 'rights-%s-%s',
+            'id_field': 'exploration_id'
+
+        }
+    }
+    # "Public" is the default value for post_commit_status for the
+    # question and skill models.
+    MODEL_NAMES_WITH_DEFAULT_COMMIT_STATUS = [
+        'QuestionSnapshotMetadataModel', 'SkillSnapshotMetadataModel']
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        job_class = AddMissingCommitLogsJob
+        return job_class.SNAPSHOT_METADATA_MODELS_WITH_MISSING_COMMIT_LOGS
+
+    @staticmethod
+    def map(snapshot_model):
+        job_class = AddMissingCommitLogsJob
+        class_name = snapshot_model.__class__.__name__
+        model_id, version_str = snapshot_model.id.rsplit('-', 1)
+        model_properties = job_class.MODEL_NAMES_TO_PROPERTIES[class_name]
+        version = int(version_str)
+        commit_log_id = (
+            model_properties['id_string_format'] % (model_id, version))
+        commit_log_model = (
+            model_properties['commit_log_model_class'].get_by_id(
+                commit_log_id))
+        commit_logs_should_exist = True
+
+        parent_model = (
+            model_properties['parent_model_class'].get_by_id(model_id))
+        if class_name == 'ExplorationRightsSnapshotMetadataModel':
+            if snapshot_model.commit_type in ['create', 'delete']:
+                commit_logs_should_exist = False
+
+        if commit_log_model is not None or not commit_logs_should_exist:
+            yield ('Found commit log model-%s' % class_name, 1)
+            return
+
+        if parent_model is None:
+            yield ('Missing Parent Model-No changes-%s' % class_name, 1)
+            return
+
+        commit_log_model = model_properties['commit_log_model_class'](
+            id=commit_log_id,
+            user_id=snapshot_model.committer_id,
+            commit_type=snapshot_model.commit_type,
+            commit_message=snapshot_model.commit_message,
+            commit_cmds=snapshot_model.commit_cmds,
+            created_on=snapshot_model.created_on,
+            last_updated=snapshot_model.last_updated,
+            version=version
+        )
+        setattr(
+            commit_log_model, model_properties['id_field'], model_id)
+        if class_name in job_class.MODEL_NAMES_WITH_DEFAULT_COMMIT_STATUS:
+            commit_log_model.post_commit_status = (
+                constants.ACTIVITY_STATUS_PUBLIC)
+        elif class_name == 'ExplorationRightsSnapshotMetadataModel':
```

Comment:
Don't hardcode. Still have a constant list and replace direct references to the string with "if X in list".

(The backend test referred to above still should exist, too.)

Reply:
Since the list had only one item, I decided to remove the list. But now I have added it back.